### PR TITLE
[Finishes #158698951] Tests for v2 API endpoints [WIP]

### PR DESCRIPTION
**Pull request by JoshuaOndieki**
**What does this PR do?**
Adds tests for v2 of the API endpoints

**Description of Task to be completed**
Endpoints should be created using the TDD standard. This will assist in creating quality endpoints with fewer or no bugs.

**What endpoints are being tested?**

EndPoint | Functionality | Note
-- | -- | --
POST `/auth/signup` | Register a user |
POST `/auth/login` | Login a user |
GET `/rides` | Fetch all available rides |
GET `/rides/<ride_id>` | Fetch the details of a single ride |
POST `/rides/<ride_id>/requests` | Make a ride request | This endpoint helps a user to indicate interest in a specific available ride.
POST `/rides` | Create a ride offer |
GET `/rides/<ride_id>/requests` | Fetch all ride requests | This endpoint helps the creator of the ride offer to see the responses so far. All the requests returned should contain the name of the interested party.
PUT `/rides/<ride_id>/requests/<request_id>` | Accept or reject a ride request.

**Are there endpoints in the IceBox?**

EndPoint | Functionality | Note
-- | -- | --
POST `/auth/logout` | Logout a user |
POST `/auth/resetpassword` | Reset a user password |
GET `/notifications` | Fetch all available notifications |
PUT `/users/<username>/edit` | Edits details of a user |
DELETE `/requests/<request_id>` | Retract a ride request
DELETE `/rides/<ride_id>` | Delete a ride offer |

**Tasks achieved in this PR**
- [x] POST `/auth/signup`
- [x] POST `/auth/login`
- [x] POST `/rides`
- [x] GET `/rides`
- [x] GET `/rides/<ride_id>`
- [x] GET `/rides/<ride_id>/requests`
- [x] PUT `/rides/<ride_id>/requests/<request_id>`

**How should one run these tests?**
- Clone this repo using `git clone https://github.com/JoshuaOndieki/ridemyway-api.git`
- Check the ReadMe for instructions on setting up the project.
- `git checkout ch-test-v2-endpoints-128698951` 
- Run `nosetests  --with-coverage --cover-package=ridemyway` to see the tests fail and coverage
**Note:** *Adding `--cover-package flag=ridemyway` will include tests for v1 that will pass.*

**Background Context?**
- These are unittests and not features.
- The build will fail.

**How can you help?**
Check the tests and suggest missing edge cases.

**Resources & Libs**
[This Pivotal Tracker board](https://www.pivotaltracker.com/n/projects/2179581) was used to plan and manage the tasks.

**Relevant PIvotal Tracker Stories?**
[#158698951](https://www.pivotaltracker.com/story/show/158698951)

### CODE REVIEW

File: ridemyway/tests/tests_v2/test_auth.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from ridemyway.tests import BaseTest
```
[the remark this answers is not in the document]
Comment (Bopchy) [reply ridemyway/tests/tests_v2/test_auth.py]: Why import BaseTest from ridemyway.tests and not ridemyway.tests_v2?

File: ridemyway/tests/tests_v2/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+"""
+    BASE testing module
+"""
+from ridemyway.tests import BaseTest
+
+
+class BaseTest(BaseTest):
```
[the remark this answers is not in the document]
Comment (Bopchy) [reply ridemyway/tests/tests_v2/__init__.py]: You have inherited this class, but not changed anything in it. If the class is going to stay as it is here, there is no difference from simply importing BaseTest like you have done for the tests above

File: ridemyway/tests/tests_v2/test_auth.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bopchy) [reply ridemyway/tests/tests_v2/test_auth.py]: Why import BaseTest from ridemyway.tests yet the tests in this file are for version 2, which I assume will be different from those in version 1? You could opt to create a second BaseTest class in ridemyway.tests that adds the setUp and tearDown specific to version 2 tests and simply import that specific class to your test files like you have done above for version 1 tests instead of inheriting it

Also, it is hard to tell which approach you want to take with your tests by simply looking at the restructuring you have done here. Could you perhaps add some code to your tests, or explain how you wish to proceed

**Review by Bopchy — CHANGES_REQUESTED** [review]
